airflow/ui/src/pages/Dag/Dag.tsx

```diff
@@ -42,20 +44,46 @@ export const Dag = () => {
     dagId,
   });
 
+  const refetchInterval = useAutoRefresh({ dagId });
+
   // TODO: replace with with a list dag runs by dag id request
   const {
     data: runsData,
     error: runsError,
     isLoading: isLoadingRuns,
   } = useDagsServiceRecentDagRuns({ dagIds: [dagId] }, undefined, {
     enabled: Boolean(dagId),
+    refetchInterval: (query) =>
+      query.state.data?.dags
+        .find((recentDag) => recentDag.dag_id === dagId)
+        ?.latest_dag_runs.some((run) => isStatePending(run.state))
+        ? refetchInterval
+        : false,
   });
 
-  const runs = runsData?.dags.find((dagWithRuns) => dagWithRuns.dag_id === dagId)?.latest_dag_runs ?? [];
+  const dagWithRuns = runsData?.dags.find((recentDag) => recentDag.dag_id === dagId);
 
   return (
     <DetailsLayout dag={dag} error={error ?? runsError} isLoading={isLoading || isLoadingRuns} tabs={tabs}>
-      <Header dag={dag} dagId={dagId} latestRun={runs[0]} />
+      <Header
+        dag={
+          dag
+            ? ({
+                ...dag,
+                // We only need to refresh latest runs and next run
+                latest_dag_runs: dagWithRuns?.latest_dag_runs,
+                next_dagrun: dagWithRuns?.next_dagrun,
+                next_dagrun_create_after: dagWithRuns?.next_dagrun_create_after,
+                next_dagrun_data_interval_end: dagWithRuns?.next_dagrun_data_interval_end,
+                next_dagrun_data_interval_start: dagWithRuns?.next_dagrun_data_interval_start,
+              } as DAGDetailsResponse & DAGWithLatestDagRunsResponse)
+            : undefined
+        }
+        dagId={dagId}
```

Comment:
Can't we retrive the dagId from the dag object  in the other props ?

Reply:
Yeah, this isn't necessary anymore since we know use other hooks inside the component. I can fix